app/assets/javascripts/miq_tree.js

```diff
@@ -147,88 +131,168 @@ function miqTreeScrollToNode(tree, id) {
-  var pieces = id.split(/-/);
-  var shortId = pieces[pieces.length - 1];
-  miqJqueryRequest('/' + ManageIQ.controller + '/snap_pressed/' + encodeURIComponent(shortId), {beforeSend: true, complete: true});
+// /////////////////////////////////////////////////////////////////////////////
+// START: ONCHECK FUNCTIONS
+// /////////////////////////////////////////////////////////////////////////////
+
+/**
+ * Function for the react tree to select checked keys.
+ * @param  {Object} tree  The tree object itself.
+ * @return {Array}        Array of keys.
+ */
+function getSelectedKeys(tree) {
+  return Object.values(tree).filter(function(entry) {
+    return (entry.state && entry.state.checked);
+  }).map(function(entry) {
+    return entry.attr.key;
+  });
 }
 
-// OnClick handler for Host Network Tree
-function miqOnClickHostNet(id) {
-  var ids = id.split('|')[0].split('_'); // Break apart the node ids
-  var nid = ids[ids.length - 1].split('-'); // Get the last part of the node id
-  DoNav('/vm_or_template/show/' + encodeURIComponent(nid[1]));
+// Generic OnCheck handler for the checkboxes in tree
+function miqOnCheckGeneric(key, checked) {
+  miqJqueryRequest(ManageIQ.tree.checkUrl + encodeURIComponent(key) + '?check=' + (checked ? '1' : '0'));
 }
 
 // OnCheck handler for the belongs to drift/compare sections tree
-function miqOnCheckSections(node, tree_name) {
-  var selectedKeys = miqTreeObject(tree_name).getChecked().map(function(n) {
-    return n.key;
-  });
+function miqOnCheckSections(key, checked, tree) {
+  var selectedKeys = getSelectedKeys(tree);
 
-  var url = ManageIQ.tree.checkUrl + '?id=' + encodeURIComponent(node.key) + '&check=' + node.state.checked;
+  var url = ManageIQ.tree.checkUrl + '?id=' + encodeURIComponent(key) + '&check=' + checked;
   miqJqueryRequest(url, {data: {all_checked: selectedKeys}});
   return true;
 }
 
-function miqOnCheckGenealogy(node, treename) {
-  var tree = miqTreeObject(treename);
-  // Map the selected nodes into an array of keys
-  var selectedKeys = tree.getChecked().map(function(item) {
-    return encodeURIComponent(item.key);
-  });
+function miqOnCheckGenealogy(key, checked, tree) {
+  var selectedKeys = getSelectedKeys(tree);
+
   // Activate toolbar items according to the selection
   miqSetToolbarCount(selectedKeys.length);
   // Inform the backend about the checkbox changes
   miqJqueryRequest(ManageIQ.tree.checkUrl + '?all_checked=' + selectedKeys, {beforeSend: true, complete: true});
 }
 
-function miqCheckAll(cb, treename) {
-  var tree = miqTreeObject(treename);
+// Services -> Catalogs -> Catalog Items -> Edit item -> Tenants tree
+function miqOnCheckTenantTree(key) {
+  sendDataWithRx({
+    controller: 'catalogItemFormController',
+    key: key,
+  });
+}
+
+/**
+ * Returns the redux tree by its name if it is in the redux store.
+ * @param  {string} treeName The tree name in the redux store.
+ * @return {object}          Tree object.
+ */
+function getReduxTree(treeName) {
```

Comment:
Do you really need this if you have just a single callsite?

Reply:
No, there were more, and I left it in because of readability, and for an occasion if it is needed again.

Reply:
I get it, but you're defining functions in the global space and you probably won't need it anywhere else :wink: